tsl/test/sql/cagg_api.sql

```diff
@@ -0,0 +1,158 @@
+  WITH  (timescaledb.continuous) AS
+SELECT time_bucket('15 minutes', time), avg(value)
+  FROM hyper_tstz
+GROUP BY 1;
+
+CREATE MATERIALIZED VIEW multi_temperature_15m
+  WITH  (timescaledb.continuous) AS
+SELECT time_bucket('15 minutes', time), avg(value)
+  FROM hyper_multi
+GROUP BY 1;
+
+SET search_path TO _timescaledb_functions, public;
+
+-- These are not part of the API, but we test them here just to make
+-- sure they work as expected.
+SELECT table_name, get_materialization_info(table_name)
+  FROM (
+      VALUES ('tstz_temperature_15m'), ('multi_temperature_15m')
+  ) t(table_name);
+
+\set ON_ERROR_STOP 0
+SELECT get_materialization_info('hyper_no_cagg');
+\set ON_ERROR_STOP 1
+
+-- This is not part of the API either, but added a test here to make
+-- sure that it works as expected.
+SELECT materialization_id,
+       to_timestamp(lowest_modified_value),
+       to_timestamp(greatest_modified_value)
+  FROM get_raw_materialization_ranges('timestamptz');
+
+-- Here are tests of the API
+SELECT *
+  INTO before
+  FROM _timescaledb_catalog.continuous_aggs_materialization_invalidation_log;
+
+CALL _timescaledb_functions.add_materialization_invalidations(
+     'tstz_temperature_15m'::regclass,
+     '["2025-04-25 11:10:00+02","2025-04-26 11:14:00+02"]'::tstzrange
+);
+
+CALL _timescaledb_functions.add_materialization_invalidations(
+     'ts_temperature_15m'::regclass,
+     '["2025-04-25 11:10:00+02","2025-04-26 11:14:00+02"]'::tsrange
+);
+
+-- Custom refresh function that iterate over the ranges inside the
+-- restriction and refresh them. This is to check that the refresh
+-- function does the right thing with the ranges returned by the API
+-- function.
+CREATE PROCEDURE custom_refresh(cagg REGCLASS, restriction ANYRANGE) AS
+$body$
+DECLARE
+  inval restriction%TYPE;
+BEGIN
+   FOR inval IN
+   SELECT UNNEST(invalidations)
+     FROM _timescaledb_functions.get_materialization_invalidations(cagg, restriction)
+   LOOP
+       RAISE NOTICE 'Updating range %', inval;
+       CALL refresh_continuous_aggregate(cagg, lower(inval), upper(inval));
+   END LOOP;
+END
+$body$
+LANGUAGE plpgsql;
+
+SELECT continuous_aggregate,
+       to_timestamp(lhs.lowest_modified_value),
+       to_timestamp(lhs.greatest_modified_value)
+  FROM _timescaledb_catalog.continuous_aggs_materialization_invalidation_log lhs
+  JOIN continuous_aggregates USING (materialization_id)
+LEFT JOIN before rhs ON row(lhs.*) = row(rhs.*)
+ WHERE lhs.materialization_id IS NULL OR rhs.materialization_id IS NULL;
+
+SELECT materialization_id,
+       to_timestamp(lowest_modified_value),
+       to_timestamp(greatest_modified_value)
+  FROM get_raw_materialization_ranges('timestamptz');
+
+SELECT * FROM _timescaledb_functions.get_materialization_invalidations(
+       'ts_temperature_15m'::regclass,
+       '["2025-04-25","2025-04-26"]'::tsrange
+);
+
+CALL custom_refresh('ts_temperature_15m', '["2025-04-25","2025-04-26"]'::tsrange);
```

Comment:
Would be also good to also check the effect of this refresh on the cagg

Reply:
I'll add that separately. It is a little more complicated and the most important part here is that the logs are updated correctly.

The custom refresh just calls normal refresh, and if that is broken, other tests will capture it.